src/query/interpreter.cpp

```diff
@@ -1521,73 +1522,156 @@ std::vector<std::vector<TypedValue>> AnalyzeGraphQueryHandler::AnalyzeGraphCreat
     const std::span<std::string> labels, DbAccessor *execution_db_accessor) {
   using LPIndex = std::pair<storage::LabelId, storage::PropertyId>;
 
-  std::vector<std::vector<TypedValue>> results;
-  std::map<LPIndex, std::map<storage::PropertyValue, int64_t>> counter;
+  std::map<LPIndex, std::map<storage::PropertyValue, int64_t>> label_property_counter;
+  std::map<LPIndex, uint64_t> vertex_degree_counter;
+  auto view = storage::View::OLD;
 
-  // Preprocess labels to avoid later checks
-  std::vector<LPIndex> indices_info = execution_db_accessor->ListAllIndices().label_property;
+  // Preprocess labels in label indexes to avoid later checks
+  std::vector<storage::LabelId> label_indices_info = execution_db_accessor->ListAllIndices().label;
   if (labels[0] != kAsterisk) {
-    for (auto it = indices_info.cbegin(); it != indices_info.cend();) {
+    for (auto it = label_indices_info.cbegin(); it != label_indices_info.cend();) {
+      if (std::find(labels.begin(), labels.end(), execution_db_accessor->LabelToName(*it)) == labels.end()) {
+        it = label_indices_info.erase(it);
+      } else {
+        ++it;
+      }
+    }
+  }
+
+  // Preprocess labels in label property indexes to avoid later checks
+  std::vector<LPIndex> label_property_indices_info = execution_db_accessor->ListAllIndices().label_property;
+  if (labels[0] != kAsterisk) {
+    for (auto it = label_property_indices_info.cbegin(); it != label_property_indices_info.cend();) {
       if (std::find(labels.begin(), labels.end(), execution_db_accessor->LabelToName(it->first)) == labels.end()) {
-        it = indices_info.erase(it);
+        it = label_property_indices_info.erase(it);
       } else {
         ++it;
       }
     }
   }
-  // Iterate over all indexed vertices
-  std::for_each(indices_info.begin(), indices_info.end(), [execution_db_accessor, &counter](const LPIndex &index_info) {
-    auto vertices = execution_db_accessor->Vertices(storage::View::OLD, index_info.first, index_info.second);
-    std::for_each(vertices.begin(), vertices.end(), [&index_info, &counter](const auto &vertex) {
-      counter[index_info][*vertex.GetProperty(storage::View::OLD, index_info.second)]++;
-    });
-  });
 
-  results.reserve(counter.size());
-  std::for_each(counter.begin(), counter.end(), [&results, execution_db_accessor](const auto &counter_entry) {
-    const auto &[label_property, values_map] = counter_entry;
+  std::vector<std::pair<storage::LabelId, storage::LabelIndexStats>> label_stats;
+  // Iterate over all label indexed vertices
+  std::for_each(label_indices_info.begin(), label_indices_info.end(),
+                [execution_db_accessor, view, &label_stats](const storage::LabelId &index_info) {
+                  auto vertices = execution_db_accessor->Vertices(storage::View::OLD, index_info);
```

Comment:
Since you are using here `storage::View::OLD` I think it makes sense to not use view variable